articles/151_node_to_participant_mapping.md

```diff
@@ -0,0 +1,155 @@
+Many `DDS` vendors don't do this (e.g.: `rti Connext` and `Fast-RTPS`), and they actually recommend creating just one `Participant` per process.
+
+### `Context`
+
+In ROS, a `Context` is the no-global state of an init-shutdown cycle.
+It also encapsulates shared state between nodes and other entities.
+In most applications, there is only one `ROS Context` in a process.
+
+## Current status
+
+There is a one-to-one mapping between `Nodes` and `DDS Participants`.
+This simplified the original implementation, as `DDS Participants` provide many features equivalent to the ones of a `ROS Node`.
+The drawback of this approach is the overhead originated by creating many participants.
+Furthermore, the maximum number of `Domain participants` is rather small.
+For example, [RTI connext](https://community.rti.com/kb/what-maximum-number-participants-domain) is limited to 120 participants per domain.
+
+## Proposed approach
+
+The goal of this proposal is to improve overall performance by avoiding the creation of one `Domain Participant` per `Node`.
+API changes will be avoided, if possible.
+
+### Mapping of `DDS Participant` to a `ROS` entity
+
+There are two alternatives, besides the current `Node` to `Participant` mapping:
+- Using one participant per process.
+- Using one participant per context.
+
+The second approach allows more flexibility, and it will allow creating more than one `Participant` for applications that need it --e.g. domain bridge applications--.
+Thus, a one to one `Participant` to `Context` mapping was chosen.
+
+In the case where multiple nodes are running in a single process, we have different options for grouping them by - ranging from a separate context for each node, over grouping a few nodes in the same context, to using a single context for all nodes.
+For most applications, only one `Context` is created.
+
+### Discovery information
+
+When not using a one to one `Node`/`Participant` mapping, extra discovery information is needed to be able to match other entities to a `Node` --e.g. publishers, subscriptions, etc--.
+Several approaches can be used to share this information, the proposed approach uses a topic for it.
+Each `Participant` publishes a message with all the information needed to match an entity to a `Node`.
+The message definition is the following:
+
+* ParticipantInfo
+  * gid
+  * NodeInfo
+    * node namespace
+    * node name
+    * reader gid
+    * writed gid
+
+When one entity is updated --e.g.: A `Publisher` is created or destroyed--, a new message is sent.
+
+Identification of `Clients` and `Services` happens according to the ros conventions for their topic names (see [	
+Topic and Service name mapping to DDS](140_topic_and_service_name_mapping.md)).
+
+This topic is considered an implementation detail, and not all `rmw` implementations have to use it.
+Thus, all the necessary logic has to be in the rmw implementation itself or in a downstream package.
```

Comment:
@ivanpauno I find this sentence awkward. Don't we conclude the exact opposite in the next two sentences?